generators/client/templates/angular/src/main/webapp/app/layouts/profiles/_page-ribbon.component.ts

```diff
@@ -4,7 +4,7 @@ import { ProfileInfo } from './profile-info.model';
 
 @Component({
     selector: '<%=jhiPrefix%>-page-ribbon',
-    template: `<div class="ribbon {{cssClass}}"><a href=""<% if (enableTranslation) { %> jhiTranslate="global.ribbon.{{ribbonEnv}}"<% } %>>{{ribbonEnv}}</a></div>`,
+    template: `<div class="ribbon" [hidden]="!ribbonEnv"><a href=""<% if (enableTranslation) { %> jhiTranslate="global.ribbon.{{ribbonEnv}}"<% } %>>{{ribbonEnv}}</a></div>`,
```

Comment:
Maybe you could use `*ngIf` rather than `[hidden]` because it removes completely the ribbon from the DOM rather than just hiding it.

`<div class="ribbon" *ngIf="ribbonEnv != null">`